optimum/intel/ipex/modeling_base.py

```diff
@@ -168,47 +84,25 @@ def __init__(
         config: PretrainedConfig = None,
         export: bool = False,
         model_save_dir: Optional[Union[str, Path, TemporaryDirectory]] = None,
-        warmup: bool = True,
         **kwargs,
     ):
-        if is_torch_xpu_available(check_device=True):
-            self._device = torch.device("xpu:0")
-        elif torch.cuda.is_available():
-            self._device = torch.device("cuda:0")
-        else:
-            self._device = torch.device("cpu")
-
-        # CPU only support jit model for now.
-        if export:
-            if isinstance(model, torch.jit.RecursiveScriptModule):
-                logger.warning("The model has been exported already.")
-            else:
-                config = model.config if config is None else config
-                use_cache = kwargs.get("use_cache", True)
-                model = ipex_jit_trace(model, self.export_feature, use_cache)
-                config.torchscript = True
-
+        config = config or model.config
         OptimizedModel.__init__(self, model=model, config=config)
 
-        self.model.to(self._device)
-        self._dtype = self.config.torch_dtype if self.config.torch_dtype is not None else torch.float32
+        self._dtype = self.model.dtype if self.model.dtype is not None else torch.float32
+        self.use_cache = kwargs.get("use_cache", False)
         self.model_save_dir = model_save_dir
-        self._is_ipex_exported = _is_patched_with_ipex(model, self.export_feature)
+        self._add_patch = _is_patched_with_ipex(model, self.export_feature, self.use_cache)
 
-        if isinstance(model, torch.jit.RecursiveScriptModule):
```

Comment:
TorchScript models will not be compatible anymore which is an important breaking change, we need to catch this to inform users

Reply:
also we need to update the documentation https://github.com/huggingface/optimum-intel/blob/ad8a4cb468b418c82d4b34dfce6ba6f1de323b18/docs/source/ipex/inference.mdx?plain=1#L18

Reply:
done.